File: 3959-check-good-integer/3959-check-good-integer.cs
public class Solution {
    public bool CheckGoodInteger(int n) {
        int ds = 0;
        int ss = 0;

        while (n > 0) {
            int curr = n % 10;
            ds += curr;
            ss += curr * curr;
            n /= 10;
        }

        return ss - ds >= 50;
    }
}

File: 3959-check-good-integer/3959-check-good-integer.rs
impl Solution {
    pub fn check_good_integer(n: i32) -> bool {
        let mut num = n;
        let mut ds = 0;
        let mut ss = 0;

        while num > 0 {
            let curr = num % 10;
            ds += curr;
            ss += curr * curr;
            num /= 10;
        }

        ss - ds >= 50
    }
}